modules/EvManager/main/car_simulatorImpl.cpp

```diff
@@ -13,14 +13,14 @@ void car_simulatorImpl::init() {
     register_all_commands();
     subscribe_to_variables_on_init();
 
+    car_simulation = std::make_unique<CarSimulation>(mod->r_ev_board_support, mod->r_ev, mod->r_slac);
+
```

Comment:
@hikinggrass I am not sure why you moved this to the init()? Publishes from other modules are handled in the `subscribe_to_variables_on_init()` in line 14.

Reply:
Looks like my commit message wasn't as clear as I intended. Yes, the subscriptions all happen in init(), using the car_simulation in the lambdas that is still a nullptr at that time (which isn't a problem during setup of the callbacks), however it can then happen that the EvManager and SlacSimulator receiver their ready() and immediately after a publish on one of these variables happens, before the car_simulation is initialized. I think this wasn't a problem before because the JsSlacSimulator was a bit slower with its first publish than the c++ version.
Moving this to init() ensures that car_simulation cannot be a nullptr at any time in ready(), which lead to a segfault in one of the ci runs (that's how I spotted it)

Reply:
Ah I see :+1: Yes, then this should definitely be moved to init()